src/components/catalog/CatalogPage.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react';
+import CatalogList from './CatalogList';
+import CatalogFilter from './CatalogFilter';
+import '../../css/catalog/CatalogPage.css';
+
+export default class CatalogPage extends Component {
+  constructor(props) {
+    super(props);
+
+    this.modalContainer = React.createRef();
+
+    this.state = {
+      selectedCourse: {},
+      loading: false,
+      courses: [{
+        title: 'JavaScript Basics',
+        categories: 'basics js frontend backend',
+        completedCourses: 2,
```

Comment:
This goes for all of your basic data and variable names in here and in some of the subcomponents: you are mixing up terminology. If one of these objects represents a course, which it does, that should have a list of lessons and maybe a count of those lessons. Not courses. Also, you probably don't need a `total_courses` variable because you'll have an array of `Lesson` objects, and can just check the length.

```suggestion
        completed_lessons: 2,
```